ISPyPSA workplan/spec for initial release

nick-gorman opened the pull request:
- A brief summary of the key requirements we plan to meet before releasing an initial version of ISPyPSA.

nick-gorman committed:
update ispypsa work plan

EllieKallmier reviewed (approved):
Looks solid to me, definitely covers the important points and I like moving the chunks down into a v1.0.0 section for follow up later. Thanks for setting this up!!

dylanjmcconnell committed:
Minor updates to work plan. Specifically added:

- Slightly more detail / context on hydro modelling
- Unit commitment and mininum generation levels for generators for v1.0.0
- Policy constraints for v1.0.0
- Small mention of introduction of monitoring / benchmarking (not tied to specific release)
